src/main/minio.js

```diff
@@ -1557,9 +1520,11 @@ export class Client {
         throw new TypeError(`response header ${header} should be of type "string"`)
       }
     })
-    var method = 'GET'
     var requestDate = new Date()
-    var query = _.map(respHeaders, (value, key) => `${key}=${uriEscape(value)}`).join('&')
+    var delimiter = '&'
+    var query = _.map(respHeaders, (value, key) => `${key}=${uriEscape(value)}`).join(delimiter)
+    if (query == '') delimiter = ''
+    if (Object.keys(reqParams).length !== 0) query = query + delimiter + querystring.stringify(reqParams)
```

Comment:
To simplify you can replace the above lines with one line:
query = querystring.stringify(Object.assign({}, reqParams, respHeaders))

Reply:
Done.